libvirt/resource_libvirt_network.go

```diff
@@ -639,22 +639,24 @@ func resourceLibvirtNetworkDelete(d *schema.ResourceData, meta interface{}) erro
 	if err != nil {
 		return fmt.Errorf("Couldn't determine if network is active: %s", err)
 	}
+	// network can be in 2 states, handles this case by case
+
+	// in case network is inactive just undefine it
 	if !active {
-		// we have to restart an inactive network, otherwise it won't be
-		// possible to remove it.
-		if err := network.Create(); err != nil {
-			return fmt.Errorf("Cannot restart an inactive network %s", err)
+		if err := network.Undefine(); err != nil {
+			return fmt.Errorf("Couldn't undefine libvirt network: %s", err)
 		}
 	}
+	// network is active, so we need to destroy it and undefine it
+	if active {
```

Comment:
What about using `else` ?

Reply:
yeah , is ok to use else also :grimacing: . Personally, subjectively I don't always like the `else` statement because you need to carry/remember the if, but depends . I mean here, is kind of subjective taste :grin: